### ISSUE TITLE
CSS declaration Autocomplete no closing semi colon

### ISSUE THREAD

**Issue by evandrocoan** [sublimehq/Packages#2076]
Originally posted on: https://github.com/SublimeTextIssues/Core/issues/2961 by @MarkCate

> ### Description: 
> Sublime Text auto-complete fails to insert CSS declaration last SEMI-COLON
> 
> ### Steps to reproduce
> 1. In a CSS declaration type any CSS property previously used, followed by a colon `:`
> 1. The autocomplete provides a space but no closing semi-colon

> ### Expected behavior
> When a CSS property is typed followed by a colon, Sublime should automatically insert a space followed by the cursor and by the semi-colon. 

> Note: When auto-complete inserts a CSS property followed by a colon and by a space, the space should also be followed by a semi-colon.

> ### Actual behavior
> When a previously used CSS property is manually typed, followed by a colon, Sublime **DOES NOT** automatically insert a space followed by the cursor and by a semi-colon. 

> Note: With fresh software, editing CSS in a new document for a variable short period of time, autocomplete works properly, with autocomplete closing each property with the space followed by the cursor and by the semi-colon. Inexplicably, after a short period of time Sublime Text begins to drop the closing semi-colon. 

> ### Environment
> * Build: 3207 Sublime Text
> * Operating system and version: macOS 10.14.6
> * 8-core 64 GB Mac Pro

> ### Wordaround
> The semi-colon insertion rules for CSS completions are defined in the CSS package. It looks like a semi-colon is never inserted when completing a property name, only when completing a property value when there is not already a semi-colon on the line.

> If you know some Python, you can edit `Packages/CSS/css_completions.py` yourself to get the behavior you desire, and if you think its likely to be useful to others and worth sharing, then you can add your patch diff to the issue.

**Comment by FichteFoll** [comment]
Could you summarize the steps for reproduction in this issue? The referenced one is a bit hard to parse.

**Comment by evandrocoan** [comment]
@FichteFoll, summarized.

**Comment by FichteFoll** [comment]
**TL;DR**: CSS property completions randomly stop inserting a semicolon at the end of the line

Unless I misunderstood something, that's the issue reported. However, I don't see how `CSS/css_completions.py` should exhibit such behavior. The logic is roughly as follows:

1. When completing a property name, add colon and space iff not in the middle of a property name already.
2. When completing a property value, add a semicolon if it's not the immediate next character.

While these conditions aren't the most refined, they are deterministic and should not exhibit seemingly "random" behavior.

**Comment by keith-hall** [comment]
sounds to me like the OP wants semi-colons to be completed when typing the property name so it completes to `prop-name: $1;` instead of `prop-name: $1`... 
if the behavior really does change based on whether the property was previously in the buffer, it would be nice to have reproduction steps. Could it be that the prop name is/was not valid/recognized so buffer completions were used instead and so no *colon* was completed/appended automatically?
